plugin.rb

```diff
@@ -246,6 +246,50 @@ class ::ListController
 
     list = list.merge(secure)
 
+    if SiteSetting.tagging_enabled
```

Comment:
Looks like this is a copy & paste from [`topic_query`](https://github.com/discourse/discourse/blob/de038c0eab576317fe7cbb647e4388fad7cc566e/lib/topic_query.rb#L693-L735), any chances you could instead use `TopicQuery`?

Reply:
Have you tried doing something like this?

```ruby
add_to_class(:topic_query, :list_messages_assigned) do |user|
  create_list(:assigned, unordered: true) do |topics|
    # here are the specific filters for the assigned topics list
    topics.where(...)
  end
end
```

Reply:
Yaa that's right but it also uses the `default_results` method to do all the filtering work. so if we re-use this method it will load all the topics and then filter assigned ones, it won't be feasible than right?

Reply:
> so if we re-use this method it will load all the topics and then filter assigned ones

Not exactly, the `default_results` method does not load all the topics. It returns an ActiveRecord object that can then be used to add more "filter" to the query before it's being executed.

Reply:
Oooh, then it will make our task much easier😅, I'll refector the code accordingly in next commit.....thanks a lot 😅

Reply:
Hey, @ZogStriP I tried this out, but this doesn't include PM, so i think we can't use this method

Reply:
Yeah you're right, `default_results` doesn't work for PMs. But now this plugin is copying so much from core... that maybe we need to try and make some changes in core to make it work better. I will take a look and see if there are any easy changes which can help us.

We have a similar problem (needing topics + PMs in the same list) for some places in core as well (e.g. bookmarks, drafts), so I think a refactor might be needed.